client/src/command.ts

```diff
@@ -0,0 +1,240 @@
+import type { ObjectToSnake } from "ts-case-convert/lib/caseConvert";
+import {
+  ConfigurationTarget,
+  Uri,
+  window,
+  workspace,
+  WorkspaceConfiguration,
+  WorkspaceFolder,
+} from "vscode";
+import { ExecuteCommandRequest } from "vscode-languageclient";
+import { LanguageClient } from "vscode-languageclient/node";
+
+const vsCodeConfigurationsObject = {
+  configurationFilePath: "",
+  debug: null,
+  tabSize: null,
+  complementAlias: null,
+  trimBindParam: null,
+  keywordCase: null,
+  identifierCase: null,
+  maxCharPerLine: null,
+  complementOuterKeyword: null,
+  complementColumnAsKeyword: null,
+  removeTableAsKeyword: null,
+  removeRedundantNest: null,
+  complementSqlId: null,
+  convertDoubleColonCast: null,
+  unifyNotEqual: null,
+} satisfies ConfigurationRecord;
+
+type ConfigurationRecord = {
+  configurationFilePath: string;
+  debug: boolean | null | undefined;
+  tabSize: number | null | undefined;
+  complementAlias: boolean | null | undefined;
+  trimBindParam: boolean | null | undefined;
+  keywordCase: string | null | undefined;
+  identifierCase: string | null | undefined;
+  maxCharPerLine: number | null | undefined;
+  complementOuterKeyword: boolean | null | undefined;
+  complementColumnAsKeyword: boolean | null | undefined;
+  removeTableAsKeyword: boolean | null | undefined;
+  removeRedundantNest: boolean | null | undefined;
+  complementSqlId: boolean | null | undefined;
+  convertDoubleColonCast: boolean | null | undefined;
+  unifyNotEqual: boolean | null | undefined;
+};
+
+// WorkspaceConfigurationを受け取り、フォーマッタで利用する設定のみのRecordにして返す
+const extractFormattingConfigurations = (
+  workspaceConfig: WorkspaceConfiguration,
+): Partial<ConfigurationRecord> => {
+  // translate null (that means unsupecified option) to undefined
+  const config: Partial<ConfigurationRecord> = {};
+  for (const key of Object.keys(vsCodeConfigurationsObject)) {
+    const value = workspaceConfig.get(key);
+    if (value != null) {
+      config[key] = value;
+    }
+  }
+  // remove configurationFilePath
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { configurationFilePath, ...restConfiguration } = config;
+
+  return restConfiguration;
+};
+
+const isFileExists = async (uri: Uri): Promise<boolean> => {
+  try {
+    await workspace.fs.stat(uri);
+    return true;
+  } catch (_) {
+    return false;
+  }
+};
+
+// uroborosql-fmt の設定ファイル名を取得する
+// ワークスペースの側で設定されている場合はその値を、設定されていない場合は ".uroborosqlfmtrc.json"を返す
+const getConfigFileName = (
+  defaultName: string = ".uroborosqlfmtrc.json",
+): string => {
+  // uroborosql-fmt の設定を取得
+  const vsCodeConfig = workspace.getConfiguration("uroborosql-fmt");
+
+  // Default value of `uroborosql-fmt.configurationFilePath` is "".
+  const vsCodeConfigPath: string = vsCodeConfig.get("configurationFilePath");
+  return vsCodeConfigPath !== "" ? vsCodeConfigPath : defaultName;
+};
+
+const getTargetFolder = (
```

Comment:
次のようにしてより共通化を進めるのはどうですか？

```js
const getTargetFolder = (): WorkspaceFolder | undefined => {
  const folders = workspace.workspaceFolders;
  if (folders === undefined) {
    // ワークスペースとしてではなく、ファイルを直接開いている場合
    window.showErrorMessage(
      "Error: Open the folder before executing commands.",
    );
    return;
  } else if (folders.length == 0) {
    // ワークスペースにフォルダが一つも存在しない場合
    window.showErrorMessage(
      "Error: There is no folder in the workspace. To execute the command, at least one folder must be added to the workspace.",
    );
    return;
  }
  if (folders.length === 1) {
    return folders[0];
  }

  // ワークスペースに複数のフォルダが存在する場合
  if (!window.activeTextEditor) {
    // activeTextEditorが存在しない場合
    return;
  }
  const activeEditorPath = window.activeTextEditor.document.uri.fsPath;

  const matchingWorkspace = folders.find((wsFolder) => {
    const relative = path.relative(wsFolder.uri.fsPath, activeEditorPath);
    return (
      relative && !relative.startsWith("..") && !path.isAbsolute(relative)
    );
  });

  if(matchingWorkspace) {
    return matchingWorkspace;
  }

  window.showErrorMessage(
    "Error: There are multiple folders in the workspace, and it could not be determined which folder's settings to target. Please select a file that belongs to the folder you want to target before executing the command.",
  );
  return;
};
```

呼び出し元の実装例：

```js
export const exportSettings = async (): Promise<void> => {
  const folder = getTargetFolder();
  if (!folder) {
    return;
  }
```